### PR TITLE
Fix Updating Link Transform

### PR DESCRIPTION
**Pull request by justism**
This MR fixes updating a kin body link transform when it is updated from webstack by requiring reinitialization.  
Currently, when the link transform is updated from webstack, the changes are reset when `SetDOFValues` is called.   
Related issues:  
https://git.mujin.co.jp/dev/webstack/-/issues/33  
FR Shanghai: https://mattermost.mujin.co.jp/china-support/pl/ojahekhhn78mpjjwsa3j67zbfo

### CODE REVIEW

**Comment by rdiankov** [comment]
@justism @ziyan i just made some changes that set the modified flags to 0xffffffff by default, and they can only be reset to 0 in the Extract call. please review.

**Comment by ziyan** [comment]
@rdiankov if you remove setting `_modifiedFields = (uint32_t)-1` in python binding, then someone who did `ExtractInfo` in python and then updating the python object, and then `UpdateFromInfo` in python, will be confused since their change will not take effect

**Comment by rdiankov** [comment]
@ziyan python binding always creates new Info, so it will always be modified now, so it will work fine.

**Comment by rdiankov** [comment]
thanks~

File: include/openrave/geometry.h [hunk 1]
```diff
@@ -491,11 +491,10 @@ class RaveTransform
         return !operator==(right);
     }
 
-    inline bool Compare(const RaveTransform<T>& rhs, T epsilon=1e-7) const {
-        T e1 = (rot-rhs.rot).lengthsqr4();
-        T e2 = (rot+rhs.rot).lengthsqr4();
-        T e = e1 < e2 ? e1 : e2;
-        return RaveSqrt((trans-rhs.trans).lengthsqr3() + e) <= epsilon;
+    /// \brief return true if any element of transform is different by mor ethan epsilon, otherwise false.
+    inline bool CompareTransform(const RaveTransform<T>& rhs, T epsilon) const {
```
Comment (Puttichai) [inline include/openrave/geometry.h:495]: @ziyan @rdiankov Is it intentional that we don't account for the fact that `quat` and `-quat` represent the exact same orientation here?

**Pull request merged**